thirdparty/CMakeLists.txt

```
@@ -3,76 +3,55 @@ include(FetchContent)
 ###################################################################################################
 # - spdlog ----------------------------------------------------------------------------------------
 
-FetchContent_Declare(
-    spdlog
-    GIT_REPOSITORY https://github.com/gabime/spdlog.git
-    GIT_TAG        v1.7.0
-    GIT_SHALLOW    true
-)
-
-FetchContent_GetProperties(spdlog)
-if(NOT spdlog_POPULATED)
-  FetchContent_Populate(spdlog)
-  add_subdirectory(${spdlog_SOURCE_DIR} ${spdlog_BINARY_DIR} EXCLUDE_FROM_ALL)
-endif()
+CPMAddPackage(
+  NAME spdlog
+  GITHUB_REPOSITORY gabime/spdlog
+  VERSION 1.7.0
+  GIT_SHALLOW TRUE
+  )
 
 ###################################################################################################
 # - thrust/cub ------------------------------------------------------------------------------------
 
 message(STATUS "Fetching Thrust")
 
-FetchContent_Declare(
-    thrust
-    GIT_REPOSITORY https://github.com/thrust/thrust.git
-    # August 28, 2020
-    GIT_TAG        52a8bda46c5c2128414d1d47f546b486ff0be2f0
+CPMAddPackage(
+  NAME thrust
+  GITHUB_REPOSITORY thrust/thrust
+  # August 28, 2020
+  GIT_TAG 52a8bda46c5c2128414d1d47f546b486ff0be2f0
+  VERSION 1.9.10mod
```

We were previously using the hash because there wasn't a release with all the fixes we needed yet. 1.10.0 was just released that has everything we need. 
```suggestion
  VERSION 1.10.0
```